src/api-client.ts

```diff
@@ -185,4 +196,36 @@ export class RevAiApiClient {
             }
         }
     }
+
+    async getCaptions(id: string, fileName = null, fileLocation = null): Promise<string> {
+        try {
+            const response = await axios.get(`/jobs/${id}/captions`, {
+                headers: { 'Accept': 'application/x-subrip' }
+            });
+
+            if (fileName) {
+                const fullPath = path.join(fileLocation || '', `${fileName}.srt`);
+                await this.writeOutputToFile(fullPath, response.data);
+            }
+            return response.data;
+        } catch (error) {
+            switch (error.response.status) {
+                case 401:
+                case 404:
+                    throw new RevAiApiError(error);
+                case 409:
+                    throw new InvalidStateError(error);
+                default:
+                    throw error;
+            }
+        }
+    }
+
+    private async writeOutputToFile(filepath: string, data: string): Promise<void> {
+        await fs.writeFile(filepath, data, (err) => {
```

Comment:
I think mixing callbacks and async await reads poorly. Best practice when using await is to surround with `try {} catch {}`.

Reply:
Although, this is entirely dependent on writeFile returning a promise.
If it does return a promise, why do you have to manually throw the error when the exception should already be thrown?